### PR TITLE
WIP Update styling conventions 

### PR DESCRIPTION
**Pull request by Wadeewee**
## What happened 🤔
Update styling and coding conventions

## Insight 👀
- Update styling by reference: https://compass.nimblehq.co/development/android/user-interface/
- Update coding conventions by reference: https://github.com/nimblehq/android-templates/pull/46

### CODE REVIEW

**Comment by Tuubz** [comment]
I think there's a misunderstanding here @Wadeewee  🙏 

The purpose of "Update our styling conventions" is not actually cleaning up our-code, rather **"How do we make sure a freshly installed IDE will include our styling conventions, so all of us and any newcomers are all aligned"**

For example: 

How did we do this for Kotlin code styling? > We exported the settings in to an `.xml` file as shown in here : https://bitbucket.org/nimblehq/the1-android/pull-requests/937

So I was hoping to do something similar for other components as well, such as: `XML code styling`, `Auto Import`, `Imports ordering`, `Max-line length`, `Add new line at and of the file`, ...

Or any other ideas are welcome as well

**Comment by Tuubz** [comment]
Also, please commit your messages with uppercase next time 🙏

**Comment by Wadeewee** [comment]
> I think there's a misunderstanding here @Wadeewee 🙏
> 
> The purpose of "Update our styling conventions" is not actually cleaning up our-code, rather **"How do we make sure a freshly installed IDE will include our styling conventions, so all of us and any newcomers are all aligned"**
> 
> For example:
> 
> How did we do this for Kotlin code styling? > We exported the settings in to an `.xml` file as shown in here : https://bitbucket.org/nimblehq/the1-android/pull-requests/937
> 
> So I was hoping to do something similar for other components as well, such as: `XML code styling`, `Auto Import`, `Imports ordering`, `Max-line length`, `Indentation`, `Add new line at and of the file`, ...
> 
> Or any other ideas are welcome as well

I will paste the file.xml in Insight and update description @Tuubz That's why I tag [WIP]

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -9,68 +9,68 @@
...
+            android:layout_height="wrap_content"
+            android:layout_marginTop="20dp"
+            android:gravity="center"
+            android:text="Hello World!"
             app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintBottom_toBottomOf="parent"/>
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
 
     </androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (Tuubz) [inline app/src/main/res/layout/activity_main.xml:75]: Can I assume you applied auto-formatting here?

If so, you can see that the the view components are ordered alphabetically, we don't want that. We want view components to be ordered from **top to down** (The view components shown at the top, should be placed at the top of the file as well)

Please confirm if this was the case, as I remember I used to have this problem as well before 🤔

Comment (Wadeewee) [reply app/src/main/res/layout/activity_main.xml:75]: Yes, I use auto-formatting here.

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -9,68 +10,68 @@
...
-            android:layout_width="40dp"
-            android:layout_height="40dp"
-            android:visibility="gone"
-            android:layout_marginTop="5dp"
-            app:layout_constraintEnd_toStartOf="@id/buttonNext"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@id/imageView"/>
-
-        <Button
+        <androidx.appcompat.widget.AppCompatButton
```
Comment (Tuubz) [inline app/src/main/res/layout/activity_main.xml:13]: Can we please revert the order of these view components? 🙏 It seems to have been swapped

**Comment by Tuubz** [comment]
Please wait with merging/approval after finishing #46

**Comment by lydiasama** [comment]
Conflict here @Wadeewee  🙋🏻

**Comment by manh-t** [comment]
This PR become hard to rebase because we have updated many changes in `kotlin` branch, and during that we have fixed some conventions as well. As discussed with @Wadeewee , we can fix the rest convention errors later when facing it. 🎬

**Pull request closed without merging**